ext/LuxorTensorPlot/src/viz_contraction.jl

```diff
@@ -0,0 +1,138 @@
+function OMEinsumContractionOrders.ein2elimination(ein::NestedEinsum{T}) where{T}
+    elimination_order = Vector{T}()
+    _ein2elimination!(ein, elimination_order)
+    return elimination_order
+end
+
+function OMEinsumContractionOrders.ein2elimination(ein::SlicedEinsum{T, NestedEinsum{T}}) where{T}
+    elimination_order = Vector{T}()
+    _ein2elimination!(ein.eins, elimination_order)
+    # the slicing indices are eliminated at the end
+    return vcat(elimination_order, ein.slicing)
+end
+
+function _ein2elimination!(ein::NestedEinsum{T}, elimination_order::Vector{T}) where{T}
+    if ein.tensorindex == -1
+        for arg in ein.args
+            _ein2elimination!(arg, elimination_order)
+        end
+        iy = unique(vcat(getiyv(ein.eins)...))
+        for ix in unique(vcat(getixsv(ein.eins)...))
+            if !(ix in iy) && !(ix in elimination_order)
+                push!(elimination_order, ix)
+            end
+        end
+    end
+    return elimination_order
+end
+
+function elimination_frame(GViz, tng::TensorNetworkGraph{TG, TL}, elimination_order::Vector{TL}, i::Int; filename = nothing, color = (0.5, 0.5, 0.5, 0.5)) where{TG, TL}
+    GViz2 = deepcopy(GViz)
+    for j in 1:i
+        id = _get_key(tng.indices_labels, elimination_order[j])
+        GViz2.vertex_colors[id] = color
+    end
+    return show_graph(GViz2, filename = filename)
+end
+
+function OMEinsumContractionOrders.viz_contraction(ein::T, args...; kwargs...) where{T <: AbstractEinsum}
+    throw(ArgumentError("Only NestedEinsum and SlicedEinsum{T, NestedEinsum{T}} have contraction order"))
+end
+
+"""
+    viz_contraction(ein::ET; locs=StressLayout(), framerate=30, filename="contraction", pathname=".", create_gif=false, create_video=true, color=(0.5, 0.5, 0.5, 0.5), show_progress=false) where {ET <: Union{NestedEinsum, SlicedEinsum}}
+
+Visualize the contraction process of a tensor network.
+
+# Arguments
+- `ein::ET`: The tensor network to visualize.
+- `locs`: The layout algorithm to use for positioning the nodes in the graph. Default is `StressLayout()`.
+- `framerate`: The frame rate of the animation. Default is 30.
+- `filename`: The base name of the output files. Default is "contraction".
+- `pathname`: The directory path to save the output files. Default is the current directory.
+- `create_gif`: Whether to create a GIF animation. Default is `false`.
+- `create_video`: Whether to create a video. Default is `true`.
+- `color`: The color of the contraction lines. Default is `(0.5, 0.5, 0.5, 0.5)`.
+- `show_progress`: Whether to show progress information. Default is `false`.
```

Comment:
Currently I did not used the `ProgressBar.jl` to avoid inducing this deps, so currently the informations are printed directly, as shown below:
```Julia
julia> viz_contraction(nested_code, filename = "a.gif", show_progress=true)
[ Info: Generating frames, 7 frames in total
[ Info: Frame 1 of 7
[ Info: Frame 2 of 7
[ Info: Frame 3 of 7
[ Info: Frame 4 of 7
[ Info: Frame 5 of 7
[ Info: Frame 6 of 7
[ Info: Frame 7 of 7
[ Info: GIF is: /var/folders/3y/xl2h1bxj4ql27p01nl5hrrnc0000gn/T/jl_KXWKi4/a.gif
"/var/folders/3y/xl2h1bxj4ql27p01nl5hrrnc0000gn/T/jl_KXWKi4/a.gif"
```
I worry that being a little bit messy.
Do you think we should add `ProgressBar.jl` somewhere or just keep the current way?